src/CommunityToolkit.Aspire.Client.MassTransit.RabbitMQ/MassTransitClientExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿using MassTransit;
+using MassTransit.Monitoring;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using System.Reflection;
+
+namespace CommunityToolkit.Aspire.Client.MassTransit.RabbitMQ;
+
+/// <summary>
+/// MassTransitClientExtensions provides extension methods for configuring MassTransit in a client application.
+/// </summary>
+public static class MassTransitClientExtensions
+{
+    /// <summary>
+    /// Configures MassTransit with RabbitMQ integration for the client side,
+    /// using the same configuration used by the hosting environment.
+    /// </summary>
+    /// <param name="services">The client service collection.</param>
+    /// <param name="name">A unique name for the RabbitMQ instance.</param>
+    /// <param name="telemetry">Enables telemetry, which could be exported to either OpenTelemetry or Application Insights.</param>
+    public static void AddMassTransitClient(this IServiceCollection services, string name, bool? telemetry = false)
+    {
+        if (string.IsNullOrWhiteSpace(name))
+            throw new ArgumentException("The name must be a valid, non-empty string.", nameof(name));
+
+        var configuration = services.BuildServiceProvider().GetRequiredService<IConfiguration>();
+        var configurationSection = configuration.GetSection($"MassTransit:{name}");
+        var options = new MassTransitOptions();
+        configurationSection.Bind(options);
+
+        services.AddMassTransit(x =>
+        {
+            x.SetKebabCaseEndpointNameFormatter();
+            x.SetInMemorySagaRepositoryProvider();
+            var entryAssembly = Assembly.GetEntryAssembly();
+            x.AddConsumers(entryAssembly);
+            x.AddSagaStateMachines(entryAssembly);
+            x.AddSagas(entryAssembly);
+            x.AddActivities(entryAssembly);
+
+            x.UsingRabbitMq((context, cfg) =>
+            {
+                cfg.Host(options.Host, options.VirtualHost, h =>
+                {
+                    if (!string.IsNullOrEmpty(options.Username))
+                        h.Username(options.Username);
+                    if (!string.IsNullOrEmpty(options.Password))
+                        h.Password(options.Password);
+                });
+
+                cfg.ConfigureEndpoints(context);
+            });
+        });
+
+        if (telemetry == true)
+        {
+            services.AddOpenTelemetry()
+                .WithMetrics(b => b
+                    .AddMeter(InstrumentationOptions.MeterName)
+                ).WithTracing(builder =>
+                {
+                    builder
+                        .AddSource(InstrumentationOptions.MeterName);
+                });
+        }
+    }
+}
+
+/// <summary>
+/// Configuration options for MassTransit client integration 
+/// </summary>
+internal sealed class MassTransitOptions
```

Comment:
This could be a public `MassTransitClientSettings` with XML docs and a `DisableTelemetry` prop.

Instead of having a `telemetry` parameter in `AddMassTransitClient` extension method, you could have an `Action<MassTransitClientSettings>` typed parameter to allow set up of some values after binding the config.